### PR TITLE
[ELY-2721] upgraded to ApacheDS AM27 and Apache Kerby KDC

### PR DESCRIPTION
**Pull request by istudens**
Issue: https://issues.redhat.com/browse/ELY-2721

### CODE REVIEW

**Comment by istudens** [comment]
@darranl @fjuma Can you review this please? Thanks

**Comment by darranl** [comment]
@istudens FYI I just enabled CI for you

File: tests/base/pom.xml [hunk 1]
```diff
@@ -642,11 +642,6 @@
             <artifactId>apacheds-kerberos-codec</artifactId>
             <scope>test</scope>
         </dependency>
-        <dependency>
-            <groupId>org.apache.directory.server</groupId>
-            <artifactId>apacheds-protocol-kerberos</artifactId>
```
Comment (fjuma) [inline tests/base/pom.xml:647]: Can this be removed from the main pom.xml as well?

File: tests/base/src/test/java/org/wildfly/security/sasl/gssapi/TestKDC.java [hunk 2]
```diff
@@ -186,7 +181,11 @@ private void stopKDC() {
             return;
         }
 
-        kdcServer.stop();
+        try {
+            kdcServer.stop();
+        } catch (KrbException e) {
+            // ignore
```
Comment (fjuma) [inline tests/base/src/test/java/org/wildfly/security/sasl/gssapi/TestKDC.java]: Might be good to throw an IllegalStateException here to be consistent with other exceptions that are caught in this test class.

**Review by fjuma — CHANGES_REQUESTED** [review]
Thanks @istudens! Have added a couple small comments.

File: tests/base/pom.xml (hunk 1 above)
Comment (istudens) [reply tests/base/pom.xml:647]: done

File: tests/base/src/test/java/org/wildfly/security/sasl/gssapi/TestKDC.java (hunk 2 above)
Comment (istudens) [reply tests/base/src/test/java/org/wildfly/security/sasl/gssapi/TestKDC.java]: done

**Comment by istudens** [comment]
@fjuma thanks for the review

**Review by fjuma — APPROVED** [review]
Thanks @istudens!

**Review by darranl — APPROVED** [review]

**Comment by darranl** [comment]
Thank you for taking on this task and looking into the Kerby changes.

**Pull request merged**